std/datetime/date.d

```diff
@@ -2869,7 +2869,13 @@ public:
     {
         import std.format : format;
         try
-            return format("%sT%s", _date.toISOString(), _tod.toISOString());
```

Comment:
Have you tried implementing toISOString with support for sinks (see e.g. the [wiki](https://wiki.dlang.org/Defining_custom_print_format_specifiers) on this)?
In theory this should allow yielding a similar performance?